src/coord/src/coord.rs

```diff
@@ -582,6 +590,44 @@ impl Coordinator {
         }: WorkerFeedbackWithMeta,
     ) {
         match message {
+            WorkerFeedback::PeekResponse(conn_id, response) => {
+                // We use an `if let` here because the peek could have been cancelled already.
+                if let Some((channel, countdown)) = self.pending_peeks.get_mut(&conn_id) {
+                    channel
+                        .send(response)
+                        .expect("Peek endpoint terminated prematurely");
+                    *countdown -= 1;
+                    if *countdown == 0 {
+                        self.pending_peeks.remove(&conn_id);
+                    }
+                }
+            }
+            WorkerFeedback::TailResponse(sink_id, response) => {
+                // We use an `if let` here because the peek could have been cancelled already.
+                // We can also potentially receive multiple `Complete` responses, followed by
+                // a `Dropped` response.
+                if let Some(channel) = self.pending_tails.get_mut(&sink_id) {
+                    match response {
+                        TailResponse::Rows(rows) => {
+                            // TODO(benesch): the lack of backpressure here can result in
+                            // unbounded memory usage.
+                            let result = channel.send(rows);
+                            if result.is_err() {
+                                // TODO(benesch): we should actually drop the sink if the
+                                // receiver has gone away. E.g. form a DROP SINK command?
+                            }
+                        }
+                        TailResponse::Complete => {
+                            // TODO: Indicate this explicitly.
+                            self.pending_tails.remove(&sink_id);
+                        }
+                        TailResponse::Dropped => {
+                            // TODO: Could perhaps do this earlier, in response to DROP SINK.
+                            self.pending_tails.remove(&sink_id);
+                        }
+                    }
```

Comment:
@benesch I imported a bunch of TODO items from `tail.rs` some of which had your name in them. It seems like, e.g. the sink dropping should be easier to perform here at the coordinator vs back at the tail operator.

Reply:
SGTM!!